vendors/shopgate_library/classes/core.php

```diff
@@ -1242,6 +1320,26 @@ protected function user_print_r($subject, $ignore = array(), $depth = 1, $refCha
 			echo $subject . "\n";
 		}
 	}
+
+	/**
+	 * @param int $memoryLimit in MB
+	 */
+	public function setExportMemoryLimit($memoryLimit) {
+		if(!empty($memoryLimit) && is_int($memoryLimit)){
+			@ini_set('memory_limit', "{$memoryLimit}M");
+		}
+	}
+
+	/**
+	 * @param int $timeLimit in seconds
+	 */
+	public function setExportTimeLimit($timeLimit) {
+
+		if(!empty($timeLimit) && (is_int($timeLimit)||is_float($timeLimit))){
+			@set_time_limit($timeLimit);
+			@ini_set('max_execution_time', $timeLimit);
+		}
+	}
```

Comment:
You do not know the server configurations on which run you module. Please remove these functions which should not be used on a production environment.